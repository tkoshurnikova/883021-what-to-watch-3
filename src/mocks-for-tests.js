export const FilmDetails = {
  title: `Whispering Mist`,
  genre: `Comedy`,
  year: 1995,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
};

export const films = [
  {
    "title": `The No Flowers`,
    "image": `http://placehold.it/280x175`,
    "description": `qwerty`,
    "rating": 1.0,
    "numberOfVotes": 1982,
    "director": `Taika Di Caprio`,
    "actors": [`Ashley Cooper`, `Ramona Flowers`],
    "genre": `Dramas`,
    "year": 1900,
    "preview": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "video": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "background_color": `#ffffff`,
    "background_image": `some image`,
    "poster_image": `some image`
  },
  {
    "title": `Rings of Cloud`,
    "image": `http://placehold.it/280x175`,
    "description": `qwerty`,
    "rating": 1.0,
    "numberOfVotes": 1982,
    "director": `Taika Di Caprio`,
    "actors": [`Ashley Cooper`, `Ramona Flowers`],
    "genre": `Dramas`,
    "year": 1900,
    "preview": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "video": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "background_color": `#ffffff`,
    "background_image": `some image`,
    "poster_image": `some image`
  },
  {
    "title": `The Sword's Memory`,
    "image": `http://placehold.it/280x175`,
    "description": `qwerty`,
    "rating": 1.0,
    "numberOfVotes": 1982,
    "director": `Taika Di Caprio`,
    "actors": [`Ashley Cooper`, `Ramona Flowers`],
    "genre": `Dramas`,
    "year": 1900,
    "preview": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "video": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "background_color": `#ffffff`,
    "background_image": `some image`,
    "poster_image": `some image`
  },
  {
    "title": `The Voyagers of the Words`,
    "image": `http://placehold.it/280x175`,
    "description": `qwerty`,
    "rating": 1.0,
    "numberOfVotes": 1982,
    "director": `Taika Di Caprio`,
    "actors": [`Ashley Cooper`, `Ramona Flowers`],
    "genre": `Dramas`,
    "year": 1900,
    "preview": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "video": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "background_color": `#ffffff`,
    "background_image": `some image`,
    "poster_image": `some image`
  },
  {
    "title": `Shores in the Heart`,
    "image": `http://placehold.it/280x175`,
    "description": `qwerty`,
    "rating": 1.0,
    "numberOfVotes": 1982,
    "director": `Taika Di Caprio`,
    "actors": [`Ashley Cooper`, `Ramona Flowers`],
    "genre": `Dramas`,
    "year": 1900,
    "preview": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "video": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "background_color": `#ffffff`,
    "background_image": `some image`,
    "poster_image": `some image`
  },
  {
    "title": `Missing Illusion`,
    "image": `http://placehold.it/280x175`,
    "description": `qwerty`,
    "rating": 1.0,
    "numberOfVotes": 1982,
    "director": `Taika Di Caprio`,
    "actors": [`Ashley Cooper`, `Ramona Flowers`],
    "genre": `Dramas`,
    "year": 1900,
    "preview": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "video": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "background_color": `#ffffff`,
    "background_image": `some image`,
    "poster_image": `some image`
  },
  {
    "title": `The Unwilling Man`,
    "image": `http://placehold.it/280x175`,
    "description": `qwerty`,
    "rating": 1.0,
    "numberOfVotes": 1982,
    "director": `Taika Di Caprio`,
    "actors": [`Ashley Cooper`, `Ramona Flowers`],
    "genre": `Dramas`,
    "year": 1900,
    "preview": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "video": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "background_color": `#ffffff`,
    "background_image": `some image`,
    "poster_image": `some image`
  },
  {
    "title": `Teacher in the Hunter`,
    "image": `http://placehold.it/280x175`,
    "description": `qwerty`,
    "rating": 1.0,
    "numberOfVotes": 1982,
    "director": `Taika Di Caprio`,
    "actors": [`Ashley Cooper`, `Ramona Flowers`],
    "genre": `Dramas`,
    "year": 1900,
    "preview": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "video": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "background_color": `#ffffff`,
    "background_image": `some image`,
    "poster_image": `some image`
  }
];
