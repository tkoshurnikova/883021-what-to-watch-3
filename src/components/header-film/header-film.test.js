import React from "react";
import renderer from "react-test-renderer";
import HeaderFilm from "./header-film.jsx";
import {FilmDetails} from "../../mocks-for-tests.js";
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';
import {Router} from "react-router-dom";
import history from "../../history";

const mockStore = configureMockStore([]);

it(`Render HeaderFilm`, () => {
  const store = mockStore({
    USER: {
      authorizationStatus: `NO_AUTH`
    }
  });
  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <HeaderFilm
              film={FilmDetails}
            />
          </Router>
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
