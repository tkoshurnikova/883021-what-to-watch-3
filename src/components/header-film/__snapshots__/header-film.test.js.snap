// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render HeaderFilm 1`] = `
<div
  className="movie-card__desc"
>
  <h2
    className="movie-card__title"
  >
    Whispering Mist
  </h2>
  <p
    className="movie-card__meta"
  >
    <span
      className="movie-card__genre"
    >
      Comedy
    </span>
    <span
      className="movie-card__year"
    >
      1995
    </span>
  </p>
  <div
    className="movie-card__buttons"
  >
    <a
      className="btn btn--play movie-card__button"
      href="/player/0"
      onClick={[Function]}
      type="button"
    >
      <svg
        height={19}
        viewBox="0 0 19 19"
        width={19}
      >
        <use
          xlinkHref="#play-s"
        />
      </svg>
      <span>
        Play
      </span>
    </a>
    <button
      className="btn btn--list movie-card__button"
      type="button"
    >
      <svg
        height={20}
        viewBox="0 0 19 20"
        width={19}
      >
        <use
          xlinkHref="#add"
        />
      </svg>
      <span>
        My list
      </span>
    </button>
    
  </div>
</div>
`;
